File: PM_Backend/Common/src/main/java/dto/ClientRequest.java
package dto;

public class ClientRequest {
    private String requestType;
    private String requestData;

    // Empty constructor required for deserialization
    public ClientRequest() {}

    public ClientRequest(String requestType, String requestData) {
        this.requestType = requestType;
        this.requestData = requestData;
    }

    public String getRequestType() {
        return requestType;
    }

    public void setRequestType(String requestType) {
        this.requestType = requestType;
    }

    public String getRequestData() {
        return requestData;
    }

    public void setRequestData(String requestData) {
        this.requestData = requestData;
    }
}

File: PM_Frontend/Shared/CommunicationObjects/ClientRequest.cs
using Shared.Dtos;

namespace Shared.CommunicationObjects;

public class ClientRequest
{
    public ClientRequest()
    {
    }

    public ClientRequest(string requestType, DTO dto)
    {
        this.requestType = requestType;
        this.dto = dto;
    }

    public string requestType { get; set; }
    public DTO dto { get; set; }
}